scalding-core/src/main/scala/com/twitter/scalding/reducer_estimation/Common.scala

```
@@ -181,6 +187,13 @@ object ReducerEstimatorStepStrategy extends FlowStepStrategy[JobConf] {
       // if still None, make it '-1' to make it simpler to log
       val numReducers = combinedEstimator.estimateReducers(info)
 
+      numReducers.foreach { n =>
+        val configuredMax = conf.getInt(EstimatorConfig.maxEstimatedReducersKey, EstimatorConfig.defaultMaxEstimatedReducers)
+        if (n > configuredMax) {
+          throw new IllegalArgumentException(s"Reducer estimator estimated $n reducers, which is more than the configured maximum of $configuredMax")
```

We had talked about this internally I think. Pig just caps at 999. The result is that almost all jobs ran with 999. The job that stumbled across this bug and got us to think about fixing things was found sort of by accident, if it had thrown an exception we would have found out sooner that there was a bug in the reducer estimator. It would be easy to just apply the cap, but generally I prefer to fail fast when expectations are not met.